src/Compiler/Checking/CheckPatterns.fs

```diff
@@ -35,6 +35,23 @@ type cenv = TcFileState
 // Helpers that should be elsewhere
 //-------------------------------------------------------------------------
 
+let inline findSynAttribute (attrName: string) (synAttrs: SynAttributes) =
+    let attributesToSearch =
+        if attrName.EndsWith("Attribute") then
+            set [attrName; attrName.AsSpan().Slice(0, attrName.Length - 9).ToString()]
```

Comment:
If you are ToString()'ing a span eventually, this could have been a regular substring / slice.

Reply:
Yeah, wanted to do something a bit more sophisticated, but didn't cleanup `AsSpan()`.